subsys/bluetooth/host/att.c

```diff
@@ -3148,6 +3148,13 @@ static void att_timeout(struct k_work *work)
 	 * target device on this ATT Bearer.
 	 */
 	bt_att_disconnected(&chan->chan.chan);
+
+	/* The timeout state is local and can block new ATT operations, but does not affect the
+	 * remote side. Disconnecting the GATT connection upon ATT timeout simplifies error handling
+	 * for developers. This reduces rare failure conditions to a common one, allowing developers
+	 * to handle unexpected disconnections without needing special cases for ATT timeouts.
+	 */
+	bt_conn_disconnect(chan->chan.chan.conn, BT_HCI_ERR_REMOTE_USER_TERM_CONN);
```

Comment:
If we are doing the `bt_conn_disconnect` here, do we still need to call to `bt_att_disconnected`?

Reply:
> Disconnect command may fail leaving ATT channel active eventually allowing an application to violate the spec

Should we actually check and log that or something?